hub/config/src/main/java/uk/gov/ida/hub/config/domain/IdentityProviderConfig.java

```diff
@@ -126,8 +130,12 @@ public Boolean isEnabledForSingleIdp() {
         return onboardingTransactionEntityIds;
     }
 
-    public boolean isOnboardingForTransactionEntity(String transactionEntity) {
-        return this.getOnboardingTransactionEntityIds().contains(transactionEntity);
+    public boolean isOnboardingForTransactionEntityAtLoa(String transactionEntity, LevelOfAssurance levelOfAssurance){
+        boolean isOnboarding = levelOfAssurance != null ?
+                isOnboardingAtLoa(levelOfAssurance) :
+                isOnboardingAtAllLevels();
+
+        return !isOnboarding || getOnboardingTransactionEntityIds().contains(transactionEntity);
```

Comment:
The naming of this function (still?) seems wrong. If we return true if no services are onboarding, thencalling it `isOnboarding`-anything feels wrong? I feel like that part of the test should be outside the function, at the caller, if necessary – or we need a rename (but to what I've no idea).

Reply:
This is a method that replaces the deleted `OnboardingForTransactionEntityPredicate` class. That predicates logical expression as a boolean method is `isOnboardingForTransactionEntity`. That was previously already being used, and the predicate also used LOA. Given that, I think this is the correct name for the method at this stage. Because all the code around this is so confusing, I'm intentionally only making changes that involve moving the code around and not changing the wording. In the next step I will look at how this can all be made clearer and more easily understood. I think this is a reasonable stage for a PR. The alternative is to do the next step and ask reviewers to understand a much larger change, with more moving pieces.